.travis.yml

```diff
@@ -19,20 +15,30 @@ env:
     - secure: "Xw7rI/qlML1nD2e2XwlakkhKAWNGZKqqE+Q3ntTvFpfHryl7KLCvVzJ4LIavnL6kGJaWOgy9vlSoEWn5g9nqHSfE31C/k5pY5nTMAKiwiJzfAS+r0asKXW2gmKhwtcTBkqyLVOZLCJSPVlFRQyfBJHY+Fs0L3KWcnMQgtBlyDhU="
   matrix:
     # The empty SCALAJS_VERSION will only compile for the JVM
-    - SCALAJS_VERSION=
-    - SCALAJS_VERSION=0.6.28
-    - SCALAJS_VERSION=1.0.0-M8
+    - SCALAJS_VERSION=         ADOPTOPENJDK=8
+    - SCALAJS_VERSION=0.6.28   ADOPTOPENJDK=8
+    - SCALAJS_VERSION=1.0.0-M8 ADOPTOPENJDK=8
+    - SCALAJS_VERSION=         ADOPTOPENJDK=11
 
 matrix:
   exclude:
-    - jdk: openjdk11
-      env: SCALAJS_VERSION=0.6.28
-    - jdk: openjdk11
-      env: SCALAJS_VERSION=1.0.0-M8
-    - scala: 0.15.0-RC1
-      env: SCALAJS_VERSION=0.6.28
-    - scala: 0.15.0-RC1
-      env: SCALAJS_VERSION=1.0.0-M8
+    - scala: 0.16.0-RC3
+      env: SCALAJS_VERSION=0.6.28   ADOPTOPENJDK=8
+    - scala: 0.16.0-RC3
+      env: SCALAJS_VERSION=1.0.0-M8 ADOPTOPENJDK=8
+
+before_install:
+  # adding $HOME/.sdkman to cache would create an empty directory, which interferes with the initial installation
+  - "[[ -d /home/travis/.sdkman/ ]] && [[ -d /home/travis/.sdkman/bin/ ]] || rm -rf /home/travis/.sdkman/"
+  - curl -sL https://get.sdkman.io | bash
+  - echo sdkman_auto_answer=true > /home/travis/.sdkman/etc/config
+  - source "/home/travis/.sdkman/bin/sdkman-init.sh"
+
+install:
+  - sdk install java $(sdk list java | grep -o "$ADOPTOPENJDK\.[0-9\.]*hs-adpt" | head -1)
+  - unset _JAVA_OPTIONS
+  - unset JAVA_HOME
```

Comment:
Could you explain this line? Thanks!

Reply:
JAVA_HOME is set by Travis CI's base image or `language: scala` to be whatevet the JDK, like JDK 11 for xenial. since I'm installing a new JDK I don't want the java on PATH and JAVA_HOME to be not inconsistent. This normally only matters when there's forking or Java only project. Alternatively I could try to do `which java` and set JAVA_HOME to wherever SDKMAN installs AdoptOpenJDK JDK.

Reply:
OK, so `sdk install` adds the installed version to the path, but doesn't set `JAVA_HOME`. I'm fine with unsetting if we don't need it. For the record, `jabba use` sets both the `PATH` and `JAVA_HOME`.